File: src/main/advisors/CXX.java
package main.advisors;

import java.io.IOException;
import java.net.URL;

public class CXX {
  public CXX() {

  }
  
  /** 
   * @return String
   * @throws IOException
   */
  public String callAPI() throws IOException {
    if(System.getProperty("os.name").contains("Windows")) {
      URL windowsAPI = getClass().getResource("/apiwrapper.exe");
      assert windowsAPI != null;
      return Host.runProcess(Runtime.getRuntime(), windowsAPI.getPath());
    }
    return null;
  }

  
  /** 
   * @return String
   * @throws IOException
   */
  public String veriyFile() throws IOException {
    if(System.getProperty("os.name").contains("Windows")) {
      URL windowsAPI = getClass().getResource("/fileint.exe");
      assert windowsAPI != null;
      return Host.runProcess(Runtime.getRuntime(), windowsAPI.getPath());
    }
    return null;
  }

  
  /** 
   * @param args
   * @throws IOException
   */
  public static void main(String[] args) throws IOException {
    new CXX().callAPI();
  }
  
}

File: kt/app/core/CXX.kt
package app.core

import kotlin.Throws
import java.io.IOException
import app.core.Host
import kotlin.jvm.JvmStatic
import app.core.CXX

/**
 * @author Jack Meng
 */
class CXX @Deprecated("") constructor() {
    /**
     * @return String
     * @throws IOException
     */
    @Deprecated("")
    @Throws(IOException::class)
    fun callAPI(): String? {
        if (System.getProperty("os.name").contains("Windows")) {
            val windowsAPI = javaClass.getResource("/apiwrapper.exe")!!
            return Host.runProcess(Runtime.getRuntime(), windowsAPI.path)
        }
        return null
    }

    /**
     * @return String
     * @throws IOException
     */
    @Deprecated("")
    @Throws(IOException::class)
    fun veriyFile(): String? {
        if (System.getProperty("os.name").contains("Windows")) {
            val windowsAPI = javaClass.getResource("/fileint.exe")!!
            return Host.runProcess(Runtime.getRuntime(), windowsAPI.path)
        }
        return null
    }

    companion object {
        /**
         * @param args
         * @throws IOException
         */
        @Deprecated("")
        @Throws(IOException::class)
        @JvmStatic
        fun main(args: Array<String>) {
            CXX().callAPI()
        }
    }
}